doc/neps/roadmap.rst

```diff
@@ -6,110 +6,112 @@ This is a live snapshot of tasks and features we will be investing resources
 
-- Rewrite numpy.org
-- Benchmarking: improve the extent of the existing suite, and run & render
-  the results as part of the docs or website.
+- Hardware: find a machine that can reliably run serial benchmarks
+- ASV produces graphs, could we set up a site? Currently at
+  https://pv.github.io/numpy-bench/, should that become a community resource?
 
-  - Hardware: find a machine that can reliably run serial benchmarks
-  - ASV produces graphs, could we set up a site? Currently at
-    https://pv.github.io/numpy-bench/, should that become a community resource?
+Website and documentation
+-------------------------
 
-Functionality outside core
---------------------------
+Our website (https://numpy.org) is in very poor shape and needs to be rewritten
+completely.
 
-Some things inside NumPy do not actually match the `Scope of NumPy`.
+The NumPy `documentation <https://www.numpy.org/devdocs/user/index.html>`__ is
+of varying quality - in particular the User Guide needs major improvements.
 
-- A backend system for `numpy.fft` (so that e.g. `fft-mkl` doesn't need to monkeypatch numpy)
+Random number generation policy & rewrite
+-----------------------------------------
 
-- Rewrite masked arrays to not be a ndarray subclass -- maybe in a separate project?
-- MaskedArray as a duck-array type, and/or
-- dtypes that support missing values
+A new random number generation framework with higher performance generators is
+close to completion, see `NEP 19`_ and `PR 13163`_.
 
-- Write a strategy on how to deal with overlap between numpy and scipy for `linalg` and `fft` (and implement it).
+Indexing
+--------
 
-- Deprecate `np.matrix`
+We intend to add new indexing modes for "vectorized indexing" and "outer indexing",
+see `NEP 21`_.
 
 Continuous Integration
 ----------------------
 
 We depend on CI to discover problems as we continue to develop NumPy before the
 code reaches downstream users.
 
-- CI for more exotic platforms (e.g. ARM is now available from
-  http://www.shippable.com/, but it is not free).
+- CI for more exotic platforms (if available as a service).
 - Multi-package testing
 - Add an official channel for numpy dev builds for CI usage by other projects so
   they may confirm new builds do not break their package.
 
-Typing
-------
+Other functionality
+-------------------
 
-Python type annotation syntax should support ndarrays and dtypes.
+- ``MaskedArray`` needs to be improved, ideas include:
 
-- Type annotations for NumPy: github.com/numpy/numpy-stubs
-- Support for typing shape and dtype in multi-dimensional arrays in Python more generally
-
-NumPy scalars
--------------
+  - Rewrite masked arrays to not be a ndarray subclass -- maybe in a separate project?
+  - MaskedArray as a duck-array type, and/or
+  - dtypes that support missing values
 
-Numpy has both scalars and zero-dimensional arrays.
+- A backend system for ``numpy.fft`` (so that e.g. ``fft-mkl`` doesn't need to monkeypatch numpy)
```

Comment:
This is a SciPy topic, do you think we should also have this in NumPy?

Reply:
This was already here, I just moved it. I agree, we may want to remove it. We now have a plan to fix this in SciPy and harmonize the APIs.

Reply:
For `scipy.fftpack`, the output of `rfft` is not complex (stored as real with a convention for 0 and nyquist); this makes it very hard to use, and would seem to make backend swapping difficult (the current docs point to `np.fft.rfft` exactly for this purpose...). I'd suggest leaving this as is for now...

Reply:
There's a plan for SciPy to create a new `scipy.fft` module with an `rfft` function matching the numpy one. That will solve that issue.

Reply:
Okay since this was already here let's leave it as is for now. But let's try not to do anything related to this in the next 4 months or so, let's first see how this works out for SciPy (there's a GSoC project on this topic).

Reply:
Great! Would be nice for the different fft packages to produce consistent output (though I very much understand the old `rfft` packing, as it allows one to do the fft in-place).